File: frontend/stock_app/lib/domain/entities/movimiento.dart
enum TipoMovimiento { entrada, salida }

class Movimiento {
  final String id;
  final String productoId;
  final String? productoNombre;
  final TipoMovimiento tipo;
  final int cantidad;
  final String? motivo;
  final String? usuarioId;
  final DateTime fecha;

  Movimiento({
    required this.id,
    required this.productoId,
    this.productoNombre,
    required this.tipo,
    required this.cantidad,
    this.motivo,
    this.usuarioId,
    required this.fecha,
  });

  bool get esEntrada => tipo == TipoMovimiento.entrada;
  bool get esSalida => tipo == TipoMovimiento.salida;

  Movimiento copyWith({
    String? id,
    String? productoId,
    String? productoNombre,
    TipoMovimiento? tipo,
    int? cantidad,
    String? motivo,
    String? usuarioId,
    DateTime? fecha,
  }) {
    return Movimiento(
      id: id ?? this.id,
      productoId: productoId ?? this.productoId,
      productoNombre: productoNombre ?? this.productoNombre,
      tipo: tipo ?? this.tipo,
      cantidad: cantidad ?? this.cantidad,
      motivo: motivo ?? this.motivo,
      usuarioId: usuarioId ?? this.usuarioId,
      fecha: fecha ?? this.fecha,
    );
  }
}

File: backend/internal/model/movimiento.go
package model

import (
	"time"
)

type TipoMovimiento string

const (
	Entrada TipoMovimiento = "entrada"
	Salida  TipoMovimiento = "salida"
)

type Movimiento struct {
	ID             string         `json:"id"`
	ProductoID     string         `json:"producto_id"`
	ProductoNombre string         `json:"producto_nombre,omitempty"`
	Tipo           TipoMovimiento `json:"tipo"`
	Cantidad       int            `json:"cantidad"`
	Motivo         *string        `json:"motivo"`
	UsuarioID      *string        `json:"usuario_id"`
	Fecha          time.Time      `json:"fecha"`
}

type CreateMovimientoRequest struct {
	ProductoID string         `json:"producto_id" validate:"required"`
	Tipo      TipoMovimiento `json:"tipo" validate:"required,oneof=entrada salida"`
	Cantidad  int            `json:"cantidad" validate:"required,min=1"`
	Motivo    *string        `json:"motivo"`
}

type MovimientoFilter struct {
	ProductoID  string
	Tipo        string
	FechaDesde  string
	FechaHasta  string
	Page        int
	Limit       int
}